pecos/xmc/xlinear/model.py

```diff
@@ -444,30 +444,49 @@ def predict(
                     Default None to disable overriding
                 post_processor (str, optional):  override the post_processor specified in the model
                     Default None to disable overriding
-                threads (int, optional): the number of threads to use for training.
+                threads (int, optional): the number of threads to use for prediction.
                     Defaults to -1 to use all
+                batch_size (int, optional): the number of instances to predict in a batch.
 
         Returns:
             Y_pred (csr_matrix): If a selected output matrix is provided, the prediction for the
                 indicated output is returned. Otherwise, a prediction matrix for some topk
                 scores is returned.
         """
-        if selected_outputs_csr is None:
-            if pred_params is None:
-                Y_pred = self.model.predict(X, pred_params=None, **kwargs)
-            elif isinstance(pred_params, self.PredParams):
-                Y_pred = self.model.predict(X, pred_params=pred_params.hlm_args, **kwargs)
+        if (pred_params is not None) and (not isinstance(pred_params, self.PredParams)):
+            raise TypeError("type(pred_kwargs) is not supported")
+
+        if "batch_size" not in kwargs:
```

Comment:
Now `kwargs = {“batch_size”: None}` will raise error. You can use `batch_size=kwargs.pop("batch_size", None)` at beggining.

Reply:
Thanks for the catch. Fixed this in the amendment.